packages/grid/src/vaadin-grid.js

```diff
@@ -473,23 +468,6 @@ class Grid extends ElementMixin(
     }
   }
 
-  /**
-   * Override an observer from `DisabledMixin` to not
-   * set `tabindex` on the grid when it is re-enabled.
-   *
-   * @param {boolean} disabled
-   * @param {boolean} oldDisabled
-   * @protected
-   * @override
-   */
-  _disabledChanged(disabled, oldDisabled) {
-    super._disabledChanged(disabled, oldDisabled);
-
-    if (oldDisabled) {
-      this.removeAttribute('tabindex');
-    }
```

Comment:
**note:** This is what caused the regression. This always removed the `tabindex` attribute after switching the grid to the enabled state which resulted in losing the attribute even when it was restored to the previously known value.